docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2305 @@
+corresponds to these similar embeddings.
+
+With that out of the way, let's walk through how the pieces fit together.
+
+
+Implementation overview
+-----------------------
+
+Broadly, the app does two things: it generates embeddings from documentation,
+and it uses those embeddings to answer user questions. The first is triggered
+manually in this implementation. We'll want to trigger it whenever the
+documentation is updated. The second is triggered automatically when the user
+asks a question.
+
+Embedding generation requires two steps:
+
+1. create embeddings for each section using `OpenAI's embeddings API
+   <https://platform.openai.com/docs/guides/embeddings>`_
+2. store the embeddings data in EdgeDB using pgvector
+
+Each time a user asks a question, our app will:
+
+1. query the database for the documentation sections most relevant to
+   the question using a similarity function
+2. inject these sections as a context into the prompt — together with user's
+   question — and submit this request to OpenAI
+3. stream the OpenAI response back to the user in real time
+
+
+Prerequisites
+=============
+
+This tutorial assumes you have `Node.js <https://nodejs.org/>`_ installed. If
+you don't, please install it before continuing.
+
+The build requires other software too, but we'll help you install it as part of
+the tutorial.
+
+.. _ref_guide_chatgpt_bot_start:
+
+
+Initial setup
+=============
+
+Let's start by scaffolding our app with the Next.js ``create-next-app`` tool.
+Run this wherever you would like to create the new directory for this project.
+
+.. code-block:: bash
+
+    $ npx create-next-app --typescript docs-chatbot
+    Need to install the following packages:
+      create-next-app@13.4.12
+    Ok to proceed? (y) y
+    ✔ Would you like to use ESLint? … No / Yes
+    ✔ Would you like to use Tailwind CSS? … No / Yes
+    ✔ Would you like to use `src/` directory? … No / Yes
+    ✔ Would you like to use App Router? (recommended) … No / Yes
+    ✔ Would you like to customize the default import alias? … No / Yes
+    Creating a new Next.js app in /<path>/<to>/<project>/docs-chatbot.
+
+Choose "Yes" for all questions except "Would you like to use \`src/\`
+directory?" and "Would you like to customize the default import alias?"
+
+Once bootstrapping is complete, you should see a success message:
+
+.. code-block::
+
+    Success! Created docs-chatbot at
+    /<path>/<to>/<project>/docs-chatbot
+
+Change into the new directory so we can get started!
+
+.. code-block:: bash
+
+    $ cd docs-chatbot
+
+Let's make a change to the ``tsconfig.json`` generated by ``create-next-app``.
+Update the ``compilerOptions`` object by setting the ``baseUrl`` property to
+the root with ``"baseUrl": "."``. Later when we add modules to the root of the
+project, this will make it easier to import them.
+
+.. lint-off
+
+.. code-block:: json-diff
+    :caption: tsconfig.json
+
+      {
+        "compilerOptions": {
+          "target": "es5",
```

Comment:
I don't know if you mentioned this later (couldn't find it with ctrl+F), we need es6 here so maybe we can show it now since we show the whole tsconfig file anyway.

Reply:
I missed this. Thank you! What's the reason for it? I will provide a brief explanation too.

Reply:
The edgedb-js needs es6 because it uses some data structures that are available from es6 onward. I know about Set() but maybe there is sth more.